onnxruntime/test/hosting/integration_tests/function_tests.py

```diff
@@ -98,5 +102,175 @@ def test_mnist_invalid_url(self):
         self.assertTrue(r.headers.get('x-ms-request-id'))
 
 
+    def test_mnist_invalid_content_type(self):
+        input_data_file = os.path.join(self.test_data_path, 'mnist_test_data_set_0_input.json')
+        url = self.url_pattern.format(self.server_ip, self.server_port, 'default_model', 12345)
+        print(url)
+
+        request_headers = {
+            'Content-Type': 'application/abc',
+            'Accept': 'application/json',
+            'x-ms-client-request-id': 'This~is~my~id'
+        }
+
+        with open(input_data_file, 'r') as f:
+            request_payload = f.read()
+
+        r = requests.post(url, headers=request_headers, data=request_payload)
+        self.assertEqual(r.status_code, 400)
+        self.assertEqual(r.headers.get('Content-Type'), 'application/json')
+        self.assertTrue(r.headers.get('x-ms-request-id'))
+        self.assertEqual(r.headers.get('x-ms-client-request-id'), 'This~is~my~id')
+        self.assertEqual(r.content.decode('utf-8'), '{"error_code": 400, "error_message": "Missing or unknown \'Content-Type\' header field in the request"}\n')
```

Comment:
I'd rather not hard code error messages because they are subject to change

Reply:
But it's fine if you want for now

Reply:
The integration test is from the customer/caller perspective.  If error message is changed, we need to update the test.